rust/arrow/src/buffer.rs

```diff
@@ -96,11 +99,18 @@ impl Debug for BufferData {
 impl Buffer {
     /// Creates a buffer from an existing memory region (must already be byte-aligned)
     pub fn from_raw_parts(ptr: *const u8, len: usize) -> Self {
+        Buffer::from_raw_parts_owned(ptr, len, true)
+    }
+
+    /// Creates a buffer from an existing memory region (must already be byte-aligned)
+    /// This differs from `from_raw_parts` in that user can provide an argument to indicate
+    /// whether this piece of memory is owned by this buffer.
+    pub fn from_raw_parts_owned(ptr: *const u8, len: usize, owned: bool) -> Self {
```

Comment:
nit: can we have another method `from_owned` which will not keep ownership of the passed in buffer? and then we can make this function private.

Also we might want to add extra doc for `from_raw_parts` indicating that it takes the ownership of the input buffer.

Reply:
ping @liurenjie1024

Reply:
Sorry for late reply since I have been on vacation. I think a better solution is to deprecate this method and add other methods with better names(from_owned_raw_parts, from_unowned_raw_parts).

Reply:
I'm -0 on deprecating the existing one merely because of a name change. On the other hand, I don't really like the name `from_owned_raw_parts` and `from_unowned_raw_parts` - they are too long and the `raw_parts` is just unnecessary. Why don't we call them `from_owned` and `from_unowned`?